tests/test_examples.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# Copyright 2020, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+""" Unit tests for examples scripts.
+
+"""
+import glob
+import os
+import shutil
+import tempfile
+import unittest
+from pathlib import Path
+
+
+class TestRepoExamples(unittest.TestCase):
+    """Unit test class for repo example scripts.
+
+    Provides a '_run_example_script' method to run (exec) a script located in
+    the 'repo example' directory.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Locate and cache repo example dir."""
+        base = Path(__file__).resolve().parents[1]
+        cls.repo_examples_dir = base / "examples" / "repo_example"
+
+    def setUp(self):
+        """Create and change into test dir.
+        NOTE: Test scripts are expected to create dirs/files in new CWD."""
+        self.original_cwd = os.getcwd()
+        self.base_test_dir = os.path.realpath(tempfile.mkdtemp())
+        os.chdir(self.base_test_dir)
+
+    def tearDown(self):
+        """Change back to original dir and remove test dir, which may contain
+        dirs/files the test created at test-time CWD."""
+        os.chdir(self.original_cwd)
+        shutil.rmtree(self.base_test_dir)
+
+    def _run_script_and_assert_files(self, script_name, filenames_created):
```

Comment:
martin is trying to enable mypy (along with the linters) in new test files so argument type annotations would be nice -- but handling these in a follow-up issue is fine IMO.

Reply:
If you really don't mind, I'll leave this to a follow-up.